### PR TITLE
refactor: switch workspace after creation

### PR DESCRIPTION
**Pull request by nivedin**
Closes HFE-483

### Description
This PR add's a improvement in the workspace creation flow where the workspace is automatically switched to the created one.

### Checks
- [x] My pull request adheres to the code style of this project
- [ ] My code requires changes to the documentation
- [ ] I have updated the documentation as required
- [x] All the tests have passed

### CODE REVIEW

File: packages/hoppscotch-common/src/components/teams/Add.vue [hunk 1]
```diff
@@ -76,8 +85,20 @@ const addNewTeam = async () => {
           // Handle GQL errors (use err obj)
         }
       },
-      () => {
+      (team) => {
         toast.success(`${t("team.new_created")}`)
+        nextTick(() => {
+          if (props.switchWorkspaceAfterCreation) {
+            if (props.switchWorkspaceAfterCreation) {
```
Comment (JoelJacobStephen) [inline packages/hoppscotch-common/src/components/teams/Add.vue:92]: We are having two same if checks here. Please remove one

**Review by JoelJacobStephen — CHANGES_REQUESTED** [review]
@nivedin When a new workspace is created, the dropdown which allows users to switch workspaces is not updated. Even after switching to the personal workspace, it is not updated.

https://github.com/hoppscotch/hoppscotch/assets/70131076/9dc8268d-b55b-49da-8f04-17bae1d98d28

File: packages/hoppscotch-common/src/components/teams/Add.vue [hunk 2]
```diff
@@ -76,8 +84,19 @@ const addNewTeam = async () => {
...
+      (team) => {
         toast.success(`${t("team.new_created")}`)
+
+        if (props.switchWorkspaceAfterCreation) {
+          REMEMBERED_TEAM_ID.value = team.id
+          workspaceService.changeWorkspace({
+            teamID: team.id,
+            teamName: team.name,
+            type: "team",
+            role: team.myRole,
```
Comment (JoelJacobStephen) [inline packages/hoppscotch-common/src/components/teams/Add.vue:96]: Resolve type issue

File: packages/hoppscotch-common/src/components/workspace/Selector.vue [hunk 3]
```diff
@@ -153,6 +157,7 @@ const switchToTeamWorkspace = (team: GetMyTeamsQuery["myTeams"][number]) => {
   workspaceService.changeWorkspace({
     teamID: team.id,
     teamName: team.name,
+    role: team.myRole,
```
Comment (JoelJacobStephen) [inline packages/hoppscotch-common/src/components/workspace/Selector.vue:160]: Resolve type issue here.

File: packages/hoppscotch-common/src/components/environments/index.vue [hunk 4]
```diff
@@ -67,7 +66,7 @@ import { GQLError } from "~/helpers/backend/GQLClient"
 import { deleteEnvironment } from "~/newstore/environments"
 import { deleteTeamEnvironment } from "~/helpers/backend/mutations/TeamEnvironment"
 import { useToast } from "~/composables/toast"
-import { WorkspaceService } from "~/services/workspace.service"
+import { MyWorkspace, WorkspaceService } from "~/services/workspace.service"
 import { useService } from "dioc/vue"
 import { Environment } from "@hoppscotch/data"
```
Comment (JoelJacobStephen) [inline packages/hoppscotch-common/src/components/environments/index.vue:71]: You can remove the unnecessary imports

**Review by JoelJacobStephen — CHANGES_REQUESTED** [review]
There are few type issues and some unnecessary imports in few components such as `SaveRequest.vue`. Rest looks good to me. Also please bring in the latest changes from `release/2024.3.3` to this PR.

**Review by JoelJacobStephen — APPROVED** [review]
LGTM!! 💯

**Review by amk-dev — APPROVED** [review]
LGTM 🚀

**Pull request merged**